### PR TITLE
ttl: add table `mysql.ttl_job_history` to store ttl job histories

### LINKED ISSUE #40654: Add table to store TTL job history

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #40654

### What is changed and how it works?

add table `mysql.ttl_job_history` to store ttl job histories

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add table `mysql.ttl_job_history`
```

### CODE REVIEW

File: ttl/ttlworker/job.go [hunk 1]
```diff
@@ -63,14 +82,44 @@ func removeTaskForJob(jobID string) (string, []interface{}) {
...
 }
 
+func addJobHistorySQL(job *ttlJob, finishTime time.Time, summaryText string) (string, []interface{}) {
+	status := cache.JobStatusFinished
+	if job.status == cache.JobStatusTimeout || job.status == cache.JobStatusCancelled {
+		status = job.status
+	}
+
+	var partitionName interface{}
+	if job.tbl.Partition.L != "" {
```
Comment (bb7133) [inline ttl/ttlworker/job.go:92]:
```suggestion
	if job.tbl.Partition.O != "" {
```
This looks better IMO.

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**